src/core/server/logging/integration_tests/rolling_file_appender.test.ts

```diff
@@ -0,0 +1,220 @@
+      expect(getFileContent('kibana.log')).toEqual(expectedFileContent([7]));
+      expect(getFileContent('kibana.1.log')).toEqual(expectedFileContent([4, 5, 6]));
+      expect(getFileContent('kibana.2.log')).toEqual(expectedFileContent([1, 2, 3]));
+    });
+
+    it('only keep the correct number of files', async () => {
+      root = createRoot({
+        kind: 'rolling-file',
+        path: logFile,
+        layout: {
+          kind: 'pattern',
+          pattern: '%message',
+        },
+        policy: {
+          kind: 'size-limit',
+          size: '60b',
+        },
+        strategy: {
+          kind: 'numeric',
+          max: 2,
+          pattern: '-%i',
+        },
+      });
+      await root.setup();
+
+      const logger = root.logger.get('test.rolling.file');
+
+      // size = 60b, message.length ~= 40b, should roll every 2 message
+
+      // last file - 'kibana-3.log' (which will be removed during the last rolling)
+      logger.info(message(1));
+      logger.info(message(2));
+      // roll - 'kibana-2.log'
+      logger.info(message(3));
+      logger.info(message(4));
+      // roll - 'kibana-1.log'
+      logger.info(message(5));
+      logger.info(message(6));
+      // roll - 'kibana.log'
+      logger.info(message(7));
+      logger.info(message(8));
+
+      await flush();
+
+      const files = readdirSync(testDir).sort();
+
+      expect(files).toEqual(['kibana-1.log', 'kibana-2.log', 'kibana.log']);
+      expect(getFileContent('kibana.log')).toEqual(expectedFileContent([7, 8]));
+      expect(getFileContent('kibana-1.log')).toEqual(expectedFileContent([5, 6]));
+      expect(getFileContent('kibana-2.log')).toEqual(expectedFileContent([3, 4]));
+    });
+  });
+
+  describe('`time-interval` policy with `numeric` strategy', () => {
+    it('rolls the log file at the given interval', async () => {
+      root = createRoot({
+        kind: 'rolling-file',
+        path: logFile,
+        layout: {
+          kind: 'pattern',
+          pattern: '%message',
+        },
+        policy: {
+          kind: 'time-interval',
+          interval: '1s',
+          modulate: true,
+        },
+        strategy: {
+          kind: 'numeric',
+          max: 2,
+          pattern: '-%i',
+        },
+      });
+      await root.setup();
+
+      const logger = root.logger.get('test.rolling.file');
+
+      const waitForNextRollingTime = () => {
+        const now = Date.now();
+        const nextRolling = getNextRollingTime(now, moment.duration(1, 'second'), true);
+        return delay(nextRolling - now + 1);
+      };
+
+      // wait for a rolling time boundary to minimize the risk to have logs emitted in different intervals
+      // the `1s` interval should be way more than enough to log 2 messages
+      await waitForNextRollingTime();
+
+      logger.info(message(1));
+      logger.info(message(2));
+
+      await waitForNextRollingTime();
```

Comment:
Integration-testing the `time-interval` policy is a pain, but I really think we do need integration tests for this appender, and I didn't want to mock anything, so I'm stuck with real time-based `await` calls. The test takes ~= 2.5sec to complete, so I think that's alright.

Reply:
can we hardcode the delay value to avoid using the internal method `getNextRollingTime`? I'd prefer we consider integration tests as 'black box' testing.

Reply:
Using a hardcoded delay would be way more flaky than properly computing the next interval tick. Best I can do would be to use `moment().startOf('second').add('second', 1)` instead of calling `getNextRollingTime`

Reply:
nah, let's keep it. `getNextRollingTime` is less cryptic.